File: 9/solution.go
package _9

func isPalindrome(x int) bool {
	if x < 0 || (x%10 == 0 && x != 0) {
		return false
	}
	y := 0
	for y < x {
		y *= 10
		y += x % 10
		x /= 10
	}
	return y == x || x == y/10
}

File: 9/solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        y = 0
        while y < x:
            y *= 10
            y += x % 10
            x //= 10
        return x == y or x == y//10
